gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/service/token/impl/TokenServiceImpl.java

```diff
@@ -263,16 +255,19 @@ private io.gravitee.am.repository.oauth2.model.AccessToken convert(JWT token, JW
     }
 
     private io.gravitee.am.repository.oauth2.model.RefreshToken convert(JWT token, User user) {
-        io.gravitee.am.repository.oauth2.model.RefreshToken refreshToken = new io.gravitee.am.repository.oauth2.model.RefreshToken();
-        refreshToken.setId(RandomString.generate());
-        refreshToken.setToken(token.getJti());
-        refreshToken.setDomain(token.getDomain());
-        refreshToken.setClient(token.getAud());
+        return convertCommon(new io.gravitee.am.repository.oauth2.model.RefreshToken(), token, user);
+    }
+
+    private <T extends io.gravitee.am.repository.oauth2.model.Token> T convertCommon(T newToken, JWT sourceToken, User user) {
+        newToken.setId(RandomString.generate());
+        newToken.setToken(sourceToken.getJti());
+        newToken.setDomain(sourceToken.getDomain());
+        newToken.setClient(sourceToken.getAud());
         // keep reference to userId in the storage, only outside world has to see the sub which maybe based on source+extId
-        refreshToken.setSubject(user == null ? token.getSub() : user.getId());
-        refreshToken.setCreatedAt(new Date(token.getIat() * 1000));
-        refreshToken.setExpireAt(new Date(token.getExp() * 1000));
-        return refreshToken;
+        newToken.setSubject(user == null ? sourceToken.getSub() : user.getFullId().lookupSubject());
```

Comment:
**issue**: please keep explicit definition using user id instead of lookup otherwise we may think the gis can be used as value which is not the case